### PR TITLE
User-specified training settings for max epochs, early stopping, learning rate.

### PR DESCRIPTION
**Pull request by kayla-leonard**
Allow user to specify the settings to use for training, and record those settings in a metadata file in output directory for quick future reference.

### CODE REVIEW

**Comment by kayla-leonard** [comment]
Solves #5

**Comment by gmwendel** [comment]
I noticed that a user can specify early_stopping_patience to be lower than learning_rate_patience. I believe this configuration would effectively disable the ReduceLROnPlateau callback, as the EarlyStopping would trigger before the learning rate has a chance to adjust.

Could you please confirm if this is the intended behavior? If not, an assert statement may need added.

**Comment by kayla-leonard** [comment]
👍 updated with assert statement
